diff --git a/makefiles/Makefile.defs b/makefiles/Makefile.defs
index 926dd33..9aab96d 100644
--- a/makefiles/Makefile.defs
+++ b/makefiles/Makefile.defs
@@ -24,3 +24,6 @@ OD			= $(TOOLS)objdump
 BUILD			?= build
 BUILD_FLAGS += -Wall -Wextra -c -Os -g -Wpointer-arith -Wl,-EL -fno-inline-functions -nostdlib -mlongcalls -mtext-section-literals -falign-functions=4 -MMD -ffunction-sections -fdata-sections
 BUILD_DEFINES = -D__ets__ -DICACHE_FLASH -U__STRICT_ANSI__ -DLWIP_OPEN_SRC -DLWIP_BUILD -DUSE_OPTIMIZE_PRINTF -DTARGET=$(target) -D$(DEFINE_TARGET)=1 -DTCP_MSS=$(TCP_MSS) -DLWIP_FEATURES=$(LWIP_FEATURES) -DLWIP_IPV6=$(LWIP_IPV6)
+
+# Additional defines for use with DuetWiFiSocketServer
+BUILD_DEFINES += -DLWIP_MDNS_RESPONDER=1 -DLINK_STATS=0 -DETHARP_STATS=0 -DIGMP_STATS=0 -DICMP_STATS=0 -DUDP_STATS=0 -DLWIP_STATS=1 -DLWIP_STATS_DISPLAY=1 -DLWIP_NETBIOS_RESPOND_NAME_QUERY=1 -DLWIP_NUM_NETIF_CLIENT_DATA=32
diff --git a/makefiles/Makefile.lwip2 b/makefiles/Makefile.lwip2
index f402cde..714439a 100644
--- a/makefiles/Makefile.lwip2
+++ b/makefiles/Makefile.lwip2
@@ -12,6 +12,9 @@ OBJ = \
 #	$(subst ../../lwip2-contrib-src/,contrib/, \
 #		$(patsubst %.c,$(BUILD)/%.o,$(wildcard ../../lwip2-contrib-src/apps/ping/*.c)))
 
+# Include addtional files for use with DuetWiFiSocketServer
+OBJ += $(patsubst %.c,$(BUILD)/%.o,$(wildcard apps/netbiosns/*.c))
+
 BUILD_INCLUDES = -I$(BUILD) -I$(SDK)/include -Iinclude -I../../glue -I../../glue-lwip -I../../glue-lwip/$(target)
 #BUILD_INCLUDES += -I../../lwip2-contrib-src/apps/ping
 
